pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -34,8 +35,129 @@ const (
 	testObservationDomainID = 0xabcd
 )
 
-// TODO: We will add another test for sendDataRecord when we support adding multiple records to single set.
-// Currently, we are supporting adding only one record from one flow key to the set.
+// TODO: Currently, we are supporting adding only one record from one flow key to the set. We will improve TestFlowAggregator_sendFlowKeyRecord when we support adding multiple records to single set.
+
+func TestFlowAggregator_sendFlowKeyRecord(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockIPFIXExpProc := ipfixtest.NewMockIPFIXExportingProcess(ctrl)
+	mockIPFIXRegistry := ipfixtest.NewMockIPFIXRegistry(ctrl)
+	mockDataSet := ipfixtest.NewMockIPFIXSet(ctrl)
+	mockRecord := ipfixtest.NewMockIPFIXRecord(ctrl)
+	mockAggregationProcess := ipfixtest.NewMockIPFIXAggregationProcess(ctrl)
+
+	fa := &flowAggregator{
+		"",
+		"",
+		"tcp",
+		nil,
+		mockAggregationProcess,
+		testExportInterval,
+		mockIPFIXExpProc,
+		testTemplateIDv4,
+		testTemplateIDv6,
+		mockIPFIXRegistry,
+		mockDataSet,
+		"",
+		nil,
+		testObservationDomainID,
+	}
+
+	testcases := []struct {
+		name       string
+		isIPv6     bool
+		flowKey    ipfixintermediate.FlowKey
+		flowRecord ipfixintermediate.AggregationFlowRecord
```

Comment:
Change this to bool for "ready to send"? You can define the flow record outside only once as it is based out of a mock and use that one instance in all testcases. This is to avoid repetition.

Reply:
Since the flowRecord type has a "ReadyToSend" field, and we want to test for both "true" and "false" for this field, I feel like we need to define it twice outside. 
So I didn't add an extra bool. I did something similar with flowKey. Hope this would also reduce repetition to some extend.